src/OrchardCore/OrchardCore.ContentManagement.Display/ContentDisplay/ContentPartDisplayDriverTPart.cs

```diff
@@ -22,44 +22,80 @@ public override ShapeResult Shape(string shapeType, Func<IBuildShapeContext, Tas
             // e.g., BodyPart.Summary, BodyPart-BlogPost, BagPart-LandingPage-Services
             // context.Shape is the ContentItem shape, we need to alter the part shape
 
-            var result = base.Shape(shapeType, shapeBuilder, initializeAsync);
+            var result = base.Shape(shapeType, shapeBuilder, initializeAsync).Prefix(Prefix);
 
+            // This should only be set in Display methods
             if (_typePartDefinition != null)
             {
-                result.Name(_typePartDefinition.Name);
+                var partName = _typePartDefinition.Name;
+                var partType = _typePartDefinition.PartDefinition.Name;
+                var contentType = _typePartDefinition.ContentTypeDefinition.Name;
 
-                result.Displaying(ctx =>
+                if (partType == shapeType)
+                {
+                    // BodyPart, Services
+                    result.Differentiator($"{partName}");
+                }
+                else
                 {
-                    // [PartType]_[DisplayType], e.g. BodyPart.Summary
-                    ctx.ShapeMetadata.Alternates.Add($"{_typePartDefinition.PartDefinition.Name}_{ctx.ShapeMetadata.DisplayType}");
+                    // ListPart-ListPartFeed
+                    result.Differentiator($"{partName}-{shapeType}");
```

Comment:
So i saw we could have e.g `BodyPart-BodyPart_Summary`. where here the shape type is `BodyPart_Summary`.

So, because there is a `-` character, with the liquid syntax we would need to use another character and then replace it in the liquid `ShapeAccessor`. We can't use `_` because as above it could be used in the shape type part.

**So here maybe use directly `_` in place of `-`.**

Also, maybe good to check if `shapeType` starts with `partName` and then not use the name twice by only using the type in this case. But maybe not good to be consistent with fields which use `{partName}-{fieldName}-{shapeType}`. Or maybe in case of a field we could check if `shapeType` starts with `fieldName`.